src/main/java/com/rapid7/client/dcerpc/mssamr/SecurityAccountManagerService.java

```diff
@@ -37,22 +37,21 @@ public SecurityAccountManagerService(RPCTransport transport) {
         this.transport = transport;
     }
 
-    public ServerHandle getServerHandle(String serverName) throws IOException {
+    public ServerHandle openServerHandle(String serverName) throws IOException {
         final SamrConnect2Request request = new SamrConnect2Request(serverName, EnumSet.of(AccessMask.MAXIMUM_ALLOWED));
         final SamrConnect2Response response = transport.call(request);
         return response.getHandle();
     }
 
-    public DomainHandle openDomain(String serverName, SID sid) throws IOException {
-        final ServerHandle handle = getServerHandle(serverName);
-        final SamrOpenDomainRequest request = new SamrOpenDomainRequest(handle, sid, EnumSet.of(AccessMask.MAXIMUM_ALLOWED));
+    public DomainHandle openDomain(ServerHandle serverHandle, SID sid) throws IOException {
+        final SamrOpenDomainRequest request = new SamrOpenDomainRequest(serverHandle, sid, EnumSet.of(AccessMask.MAXIMUM_ALLOWED));
         final SamrOpenDomainResponse response = transport.call(request);
         return response.getHandle();
     }
 
-    public DomainHandle openDomain(ServerHandle serverHandle, SID sid) throws IOException {
-        final SamrOpenDomainRequest request = new SamrOpenDomainRequest(serverHandle, sid, EnumSet.of(AccessMask.MAXIMUM_ALLOWED));
-        final SamrOpenDomainResponse response = transport.call(request);
+    public GroupHandle openGroup(DomainHandle domainHandle, int groupRID) throws IOException {
+        final SamrOpenGroupRequest request = new SamrOpenGroupRequest(domainHandle, (int) AccessMask.MAXIMUM_ALLOWED.getValue(), groupRID);
```

Comment:
FYI - the legacy code using this `AccessMask (0x00000010)`, is it working when you tested? 
` SamOpenDomainGroupRpcRequest openReq = getSamOpenDomainGroupRpcRequest(
               domHandle,
               cifsSID.getSubAuthorityIDs()[cifsSID.getSubAuthorityCount() - 1],
               0x00000010);`